lib/tax_tribunal/login.rb

```diff
@@ -0,0 +1,66 @@
+require 'oauth2'
+require 'pry'
+
+module TaxTribunal
+  class Login < Downloader
+    ORGANISATION = ENV.fetch('MOJSSO_ORG')
+    ROLE = ENV.fetch('MOJSSO_ROLE')
+    CALLBACK_URI = ENV.fetch('MOJSSO_CALLBACK_URI')
+    TOKEN_REDIRECT_URI = ENV.fetch('MOJSSO_TOKEN_REDIRECT_URI')
+
+    get '/login' do
+      if session[:email].nil?
+        redirect oauth_client.auth_code.authorize_url(redirect_uri: CALLBACK_URI)
+      else
+        session[:already_logged_in] = "You are already logged in as #{session[:email]}."
+        erb :root
+      end
+    end
+
+    get '/logout' do
+      session.destroy
+      erb :root
+    end
+
+    get '/oauth/callback' do
+      resp = authorise!(params[:code])
```

Comment:
You could combine these two lines as `if resp = [whatever]` but that's a matter of taste, I guess